File: backend/src/app/models/assignment.ts
/**
 * @file assignment.ts
 * @description
 * Assignment related models
 *
 * @module models
 * @version 1.0.0
 * @auth Thomas
 */
type Assignment = {
  name: string;
  id: number;
  createdAt: Date;
  updatedAt: Date;
  description: string;
  courseId: number;
  fileUrl: string | null;
  dueDate: Date | null;
  deletedAt: Date | null;
};

export type { Assignment };

File: SchoolSpace-Server/src/app/models/core/Assignment.cs
namespace backend.app.models.core
{
    public class Assignment : ITimestamped
    {
        public int Id { get; set; }
        public required string Title { get; set; }
        public string? Description { get; set; }
        public DateTime? DueDate { get; set; }
        public int MaxPoints { get; set; }
        public int CourseId { get; set; }
        public Course Course { get; set; } = null!;
        public ICollection<Submission> Submissions { get; set; } = [];
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}
